src/main/java/uk/gov/pay/api/filter/RateLimiterFilter.java

```diff
@@ -62,6 +62,10 @@ private void setTooManyRequestsError() throws IOException {
 
     @Override
     public void filter(ContainerRequestContext requestContext) throws IOException {
+        if ("healthcheck".equals(requestContext.getUriInfo().getPath())) {
```

Comment:
is it being overly defensive to actually do a null check on the return value of `getUriInfo` before calling `getPath`? Every request to publicapi goes through this so it should be robust but would `UriInfo` every be `null`?
```
if (requestContext.getUriInfo() != null && "healthcheck".equals(requestContext.getUriInfo().getPath())) {
```

Reply:
I think `requestContext.getUriInfo()` should never return null. Looking at the methods on `[UriInfo](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/UriInfo.html)` shows it has methods like `getAbsolutePath()` and there’s always an absolute path, so it makes sense there will always be a `UriInfo` to contain it.

Also, if we look at the source for the `ContainerRequestContext` in Glassfish, we can see `getUriInfo()` is defined like this:

```java
@Override
public ExtendedUriInfo getUriInfo() {
    return uriRoutingContext;
}
```

`uriRoutingContext` is a field initialised in the constructor:

```java
this.uriRoutingContext = new UriRoutingContext(this);
```

Nothing else in the class seems to mutate it and there’s other code in there that seems to assume it’s not null.